lib/user.rb

```diff
@@ -0,0 +1,54 @@
+require 'data_mapper'
+require 'bcrypt'
+
+class User
+
+  include DataMapper::Resource
+
+  property :id, Serial
+  property :email, String
+  property :password, Text
+  property :username, String
+  property :name, String
+
+  has n, :peeps
+
+  def self.make(email:, password:, username:, name:, encrypter: BCrypt::Password)
+    return false unless new_email?(email)
+    return false unless new_username?(username)
+    encrypted_password = encrypter.create(password)
+    create(email: email, password: encrypted_password, username: username, name: name)
+  end
+
+  def self.sign_in(username:, password:)
+    return false unless username_exist?(username)
+    return false unless password_correct?(username, password)
+    @current_user = first(username: username)
+  end
+
+  def self.sign_out
+    @current_user = nil
+  end
+
+  def self.current_user
+    @current_user
+  end
+
+  def self.new_email?(email)
```

Comment:
I feel like some of these methods should be private. Also, though, do you need them all? Each one makes another call to the database. If you just try to find a user with the username and matching encrypted password, then either there is one or there isn't, so you have one call to the database which results in a user being generated or an error if there isn't a match.